docs/RFC/K23SI_testing.md

```diff
@@ -76,10 +76,88 @@ Notes:
+```
+
+### Test cases
+
+| category                 | test case                                                    | Expected result                        | Possible fix needed |
+| ------------------------ | ------------------------------------------------------------ | -------------------------------------- | ------------------- |
+| fake-read-write conflict | Earlier Read txn encounters a WI, two txns have the same priority and tsoId | OK (read prior value)                  |                     |
+| priority of Push()       | Txn with higher priority encounters a WI                     | created (challenger won the Push)      |                     |
+| priority of Push()       | Txn with lower priority encounters a WI                      | AbortConflict (incumbent won the Push) |                     |
+| timestamp of Push()      | Earlier WRITE Txn encounters a WI with the same priority     | AbortConflict (incumbent won the Push) |                     |
+| timestamp of Push()      | Older READ/WRITE Txn encounters a WI with the same priority  | created (challenger won the Push)      |                     |
+| tsoId of Push()          | WRITE Txn encounters a WI with the same priority and timestamp, but its tso ID is smaller | AbortConflict (incumbent won the Push) |                     |
+| tsoId of Push()          | WRITE Txn encounters a WI with the same priority and timestamp, but its tso ID is bigger | created (challenger won the Push)      |                     |
+
 ## Scenario 06 - finalization
+
+### Test setup
+
+- start a cluster and assign collection. 
+- Start a transaction that have not been End().
+
+### Test cases
+
+
+| test case                                                    | Expected result                                              | Possible fix needed |
+| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------- |
+| Finalize is a non-exist record in this transaction           | OperationNotAllowed                                          |                     |
+| Finalize_Commit partial record within this transaction       | OK                                                           |                     |
+| Finalize a record whose status is commit                     | OperationNotAllowed                                          |                     |
+| After Finalize_Commit partial record, other transactions read the record while this transaction is still in progress | OK                                                           |                     |
+| After partial Finalization_commit, txn continues and then End_Commit all records | OK                                                           |                     |
+| After partial Finalization_commit, txn continues and then End_Abort all records | S500_server caught exception processing request (partial finalize success) | need fix ？         |
```

Comment:
You can add a note here that it is expected that the data will be in an inconsistent state. An S500 is ok as long as the server is not crashing and can continue to process new transactions.